internal/wasm/host.go

```diff
@@ -156,6 +172,21 @@ func addFuncs(
 		if hf, ok := v.(*HostFunc); ok {
 			nameToFunc[hf.Name] = hf
 			funcNames = append(funcNames, hf.Name)
+		} else if pf, ok := v.(*ProxyFunc); ok {
+			proxiedIdx := len(funcNames)
+			if proxiedIdx > 255 {
+				return errors.New("TODO: proxied funcidx larger than one byte")
+			}
+			nameToFunc[pf.Proxied.Name] = pf.Proxied
+			funcNames = append(funcNames, pf.Proxied.Name)
+
+			proxyBody := make([]byte, len(pf.Proxy.Code.Body))
+			copy(proxyBody, pf.Proxy.Code.Body)
+			proxyBody[pf.CallBodyPos] = byte(proxiedIdx)
```

Comment:
shoulnd't this be encoded as unsigned 32-bit leb128 ?

Reply:
I was going to limit it to what fits in one byte. I'll panic if over noting the msg thing you mentioned. There aren't enough host functions to go over a byte even doubled.